libs/execution/include/hpx/execution/executors/timed_executors.hpp

```diff
@@ -90,7 +84,7 @@ namespace hpx { namespace parallel { namespace execution {
                         .get())
             {
                 auto predecessor = make_ready_future_at(abs_time);
-                return execution::then_execute(sequenced_executor(),
+                return execution::then_execute(exec,
```

Comment:
The old code was as far as I can tell an optimization (reduces the number of tasks spawned). The new behaviour works, but leads to changes in the counts here: https://github.com/STEllAR-GROUP/hpx/blob/9600265fab389b32a8966fe3a7c476f832ec6fe1/libs/execution/tests/unit/minimal_timed_async_executor.cpp#L276-L281. That's in principle harmless. Does someone think this change is going badly harm performance? An alternative would be to move all of the timed execution functionality yet another separate module, which could depend on the new `executors` module. Thoughts?

Reply:
There was one other test that broke as a result of this change: `suspend_thread_timed`. The fix is explained in the [commit message](https://github.com/STEllAR-GROUP/hpx/pull/4494/commits/627ce6107fccbdf9f8acdff2ce0f1113ab2e44e7). I'm not convinced there isn't a better way to fix this though. @hkaiser any bright ideas?

Reply:
> There was one other test that broke as a result of this change: suspend_thread_timed. The fix is explained in the commit message. I'm not convinced there isn't a better way to fix this though. @hkaiser any bright ideas?

@msimberg this commit does not exist anymore. Could you copy the message here for posterity, please?

Reply:
Yeah, sorry about that. I went back to move the timed functionality into a separate module. I've archived the old state of the branch [here](https://github.com/msimberg/hpx/commits/module_executors_timed_changes), [this](https://github.com/msimberg/hpx/commit/ae699e435811b91e7bb1dc69f9b8337f88f03a7e) is the commit, and just in case I wipe that branch the commit message was:
```
Changes to timed_executor to eagerly create timed threads lead to the
suspend_thread_timed segfaulting for schedulers that don't support suspension.
This happened because when an exception is thrown trying to suspend a worker
thread, the thread pool gets shut down immediately, but the timed threads are
still waiting to spawn. When they wake up and try to schedule a thread on the
thread pool the thread pool is gone and the test segfaults.
```

Reply:
I'll mark this as resolved now since the change isn't relevant anymore.